primitives/state-machine/src/overlayed_changes.rs

```diff
@@ -498,6 +510,44 @@ impl OverlayedChanges {
 		)
 	}
 
+	/// Get an iterator over all pending and committed child tries in the overlay.
+	pub fn child_infos(&self) -> impl IntoIterator<Item=&ChildInfo> {
+		self.prospective.children_default.iter()
+			.chain(self.committed.children_default.iter())
+			.map(|(_, v)| &v.1).collect::<BTreeSet<_>>()
+	}
+
+	/// Get an iterator over all pending and committed changes.
+	///
+	/// Supplying `None` for `child_info` will only return changes that are in the top
+	/// trie. Specifying some `child_info` will return only the changes in that
+	/// child trie.
+	pub fn changes(&self, child_info: Option<&ChildInfo>)
```

Comment:
I like the idea of an api using `Option<childinfo>` (got similar code in some variant of my prs), note that you could also remove option and use child_info.is_top_trie (not sure if the fonction currently existr but it is in many of my prs).
Then you got choice to define top trie as a separate child type, but in my previous pr I just did define it as the default child trie located at empty key location (it removes the ability to define a default child trie at such location, but such child trie key is invalid for a default child trie anyway).

Reply:
Ok, please forget this comment, I realize that using Option is something I probably have wrote at some point to avoid making top trie specifically a child trie, it is out of scope of this pr.